hilti/toolchain/include/compiler/coercer.h

```diff
@@ -74,31 +74,32 @@ enum class CoercionStyle {
      * Shortcut style activating all possible coercions in the context of an
      * assignment.
      */
-    TryAllForAssignment = (1U << 0U) | (1U << 3U) | (1U << 4U) | (1U << 5U) | (1U << 6U) | (1U << 9U),
+    TryAllForAssignment =
+        Assignment | TryExactMatch | TryConstPromotion | TryCoercion | TryCoercionWithinSameType | TryDeref,
 
     /**
      * Shortcut style activating all possible coercions in the context of
      * operator resolution.
      */
-    TryAllForMatching = (1U << 1U) | (1U << 3U) | (1U << 4U) | (1U << 5U) | (1U << 6U),
+    TryAllForMatching = TryExactMatch | TryConstPromotion | TryCoercion | TryCoercionWithinSameType,
 
     /**
      * Shortcut style activating possible coercions in the context of
      * function parameter passing, however without allowing any type changes.
      */
-    TryDirectMatchForFunctionCall = (1U << 2U) | (1U << 3U) | (1U << 4U) | (1U << 6U),
+    TryDirectMatchForFunctionCall = FunctionCall | TryExactMatch | TryConstPromotion,
 
     /**
      * Shortcut style activating all possible coercions in the context of
      * function parameter passing.
      */
-    TryAllForFunctionCall = (1U << 2U) | (1U << 3U) | (1U << 4U) | (1U << 5U) | (1U << 6U) | (1U << 9U),
+    TryAllForFunctionCall = TryDirectMatchForFunctionCall | TryCoercion | TryCoercionWithinSameType | TryDeref,
 
     /**
      * Shortcut style allowing for direct matches only in the context of
      * operator resolution.
      */
-    TryDirectForMatching = (1U << 1U) | (1U << 3U) | (1U << 4U) | (1U << 6U)
+    TryDirectForMatching = TryExactMatch | TryConstPromotion | TryCoercionWithinSameType
```

Comment:
I don't think we need `TryCoercionWithinSameType` here (else the bit should probably not named "Direct"), can you check why we added it here? Nothing in the test suite seems to depend on it.

Reply:
Turns out it looks like we don't need `TryCoercionWithinSameType` for any of the short-cut styles anymore (mostly because `TryCoercion` already implies it). I've cleaned that up.